Document/0x04f-Testing-Network-Communication.md

```diff
@@ -205,67 +205,123 @@ If you want to verify whether your server supports the right cipher suites, ther
+## MASTG-TECH: Passive Eavesdropping
+
+This method involves passively capturing network traffic using tools such as Wireshark or tcpdump (@MASTG-TOOL-0081, @MASTG-TOOL-0080, or @MASTG-TOOL-0075). It is useful for identifying network endpoints, analyzing protocol metadata, and understanding how an app communicates with its server. However, it cannot automatically decrypt TLS-encrypted communication. That said, [TLS decryption is possible](https://wiki.wireshark.org/TLS#tls-decryption) if you can [obtain the pre-master secret](https://wiki.wireshark.org/TLS#using-the-pre-master-secret). For an example specific to Android, see [this article](https://nibarius.github.io/learning-frida/2022/05/21/sniffing-tls-traffic).
+
+### When Is it Useful?
+
+Passive eavesdropping is particularly useful in the following scenarios:
+
+- When mobile apps use cross-platform frameworks such as [Xamarin](https://www.xamarin.com/platform "Xamarin") that ignore system proxy settings.
+- When mobile apps detect and prevent the use of system proxies.
+- When intercepting push notifications, such as Google Cloud Messaging (GCM)/Firebase Cloud Messaging (FCM) on Android.
+- When monitoring XMPP or other non-HTTP-based communication protocols.
+
+### How Does It Work?
+
+Route the traffic from the mobile device to the host computer:
+
+1. Configuring the host computer as a network gateway using built-in internet sharing options on Windows, macOS, and Linux.
+2. Configure the mobile device to route HTTP(S) requests to your proxy.
+     - **Android (see @MASTG-TECH-0010)**: Use adb to forward traffic to your host computer.  
+     - **iOS (see @MASTG-TECH-0062)**: On macOS, create a "Remote Virtual Interface" to capture all traffic from an iOS device.
 
-## Intercepting Traffic on the Network Layer
+Once traffic is routed through your host, use tools like Wireshark or tcpdump to inspect and analyze it.
 
-Dynamic analysis by using an interception proxy can be straight forward if standard libraries are used in the app and all communication is done via HTTP. But there are several cases where this is not working:
+## MASTG-TECH: Achieving a MITM Position via ARP Spoofing
 
-- If mobile application development platforms like [Xamarin](https://www.xamarin.com/platform "Xamarin") are used that ignore the system proxy settings;
-- If mobile applications verify if the system proxy is used and refuse to send requests through a proxy;
-- If you want to intercept push notifications, like for example GCM/FCM on Android;
-- If XMPP or other non-HTTP protocols are used.
+When proxy-based interception fails due to non-HTTP protocols or proxy-aware apps, **ARP Spoofing** can be used to redirect network traffic. ARP Spoofing is a **Layer 2 attack** that allows an attacker to impersonate the network gateway, forcing the mobile device to send its traffic through the attacker's machine.
```

Comment:
```suggestion
When proxy-based interception fails due to non-HTTP protocols or non-proxy-aware apps, **ARP Spoofing** can be used to redirect network traffic. ARP Spoofing is a **Layer 2 attack** that allows an attacker to impersonate the network gateway, forcing the mobile device to send its traffic through the attacker's machine.
```
(I actually never use ARP spoofing, as the alternatives are much nicer to me)

Reply:
I agree! And with these changes we clarify that and lead people to the right and most useful techniques.